### ISSUE TITLE
Analog generating awful code

### ISSUE THREAD

**Issue by darthscsi** [llvm/circt#740]
We need to aggressively check if firrtl.analog arguments are just passed to modules and not generate intermediate wires or aliases for them.

**Comment by darthscsi** [comment]
This should be a direct passthrough:
```
firrtl.circuit "foo740" {
  firrtl.extmodule @bar740(%led_0: !firrtl.analog<1>)
  firrtl.module @foo740(%led_0: !firrtl.analog<1>) {
    %result = firrtl.instance @bar740 {name = "fpga", portNames = ["led_0"]} : !firrtl.analog<1>
    firrtl.attach %result, %led_0 : !firrtl.analog<1>, !firrtl.analog<1>
  }
}
```

**Comment by lattner** [comment]
Fixed in 03e3671a thx
